bin/ch/stdafx.h

```diff
@@ -152,32 +152,113 @@ do { \
 #include "JITProcessManager.h"
 #endif
 
-template<class T, bool JSRTHeap>
-class AutoStringPtr
+class AutoString
 {
-    T* data;
+    size_t length;
+    char* data;
+    LPWSTR data_wide;
+    JsErrorCode errorCode;
+    bool dontFree;
 public:
-    AutoStringPtr():data(nullptr) { }
-    ~AutoStringPtr()
+    AutoString():length(0), data(nullptr),
+        data_wide(nullptr), errorCode(JsNoError), dontFree(false)
+    { }
+
+    AutoString(JsValueRef value):length(0), data(nullptr),
+        data_wide(nullptr), errorCode(JsNoError), dontFree(false)
     {
-        if (data == nullptr)
+        Initialize(value);
+    }
+
+    JsErrorCode Initialize(JsValueRef value)
+    {
+        JsValueRef strValue;
+        JsValueType type;
+        ChakraRTInterface::JsGetValueType(value, &type);
+        if (type != JsString)
+        {
+            errorCode = ChakraRTInterface::JsConvertValueToString(value, &strValue);
+        }
+        else
+        {
+            strValue = value;
+        }
+        if (errorCode == JsNoError)
         {
-            return;
+            size_t len = 0;
+            errorCode = ChakraRTInterface::JsWriteStringUtf8(strValue, nullptr, 0, &len);
```

Comment:
This is fine, though we have JsWriteStringUtf16. In case the user instantiate this class only to GetWideString(), should we consider delay reading string content and also exercise JsWriteStringUtf16? May be a good chance to test out that API.